runtime/compiler/optimizer/SPMDParallelizer.cpp

```diff
@@ -1441,6 +1441,30 @@ bool TR_SPMDKernelParallelizer::processSPMDKernelLoopForSIMDize(TR::Compilation
             {
             // increment of PIV
             traceMsg(comp, "Reducing the number of iterations of the loop %d at storeNode [%p] by vector length %d \n",loop->getNumber(), storeNode, unrollCount);
+	    TR_ASSERT_FATAL(storeNode->getFirstChild()->getOpCode().isAdd() || storeNode->getFirstChild()->getOpCode().isSub(), "PIV increment should be simple (either by add or by sub");
+	    TR_ASSERT_FATAL(storeNode->getFirstChild()->getFirstChild()->getOpCode().isLoad(), "PIV increment should have load");
+	    TR_ASSERT_FATAL(storeNode->getFirstChild()->getSecondChild()->getOpCode().isLoadConst(), "PIV increment should have const increment value");
+            if (storeNode->getFirstChild()->getReferenceCount() > 1)
```

Comment:
Maybe I'm missing something, but if the increment is commoned with other uses why does uncommoning just the definition let the vectorization do the right thing but the commoned version does not?

Reply:
Lets says we want to vectorize the following
`A[i] = B[i+1] * 2`
Now when we do that, the i + 1 calculation in the address is used for incrementing i. Hence, they become common. Now, if we don't uncommon the i+1 node, then later on loop reduction will change the 1 to 4 (for example). Hence making i = i + 4. Now, as i + 1 was previously commoned, this change will make the code like below -
`A[i] = B[i+4] * 2`

Reply:
Also, I believe this fix should be able to fix a previous bug #9446

Reply:
Also mixed tabs/spaces in this commit should be fixed up.

Reply:
The tabs/spaces are fixed, it was introduced due to vim not having proper tab extend setup